File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> previousInt = new HashSet<>();
        int maxCount = 0;
        for(int num : nums){
            int count = 1;
            previousInt.add(num);
            int low = 1;
            while(previousInt.contains(num-(low++))) count++;

            int high = 1;
            while(previousInt.contains(num+(high++)))count++;

            maxCount = Math.max(count, maxCount);
            }
        return maxCount;
    }
}

File: Data Structures & Algorithms/majority-element-ii/submission-2.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        """
        [5,2,3,2,2,2,2,5,5,5]


        """
        dic = defaultdict(int)

        for num in nums:
            dic[num] += 1
            if len(dic) == 3:
                new_count = defaultdict(int)
                for key, value in dic.items():
                    if value > 1:
                        new_count[key] = value - 1
                dic = new_count

        res = []
        for n in dic:
            if nums.count(n) > len(nums) // 3:
                res.append(n)
        return res
